watch-library/hardware/watch/watch_hpt.h

```diff
@@ -0,0 +1,68 @@
+#ifndef WATCH_HPT_H__
+#define WATCH_HPT_H__
+
+#include <inttypes.h>
+
+/**
+ * Defines the reasons the HPT callback is being invoked. More than one flag may be set.
+*/
+typedef struct {
+    /**
+     * The callback is being invoked because of an error in the comparison (?)
+    */
+    bool cmp_error :1;
```

Comment:
What conditions could possibly cause this?

Reply:
Not sure, but it was something the datasheet called out as a possible interrupt source. Unless I read the datasheet wrong, the timer only has a single interrupt vector and you're supposed to check the interrupt status register to see what caused it.

>Bit 1 – ERR: Error Interrupt Flag
This flag is set when a new capture occurs on a channel while the corresponding Match or Capture
Channel x interrupt flag is set, in which case there is nowhere to store the new capture.

It's entirely irrelevant to the HPT though, so maybe I should get rid of the flag in this struct.

Reply:
Match or capture? This has something to do with using these timers to generate waveforms. I don't think we'll be using that functionality. It could be safe to ignore that possibility.